### PR TITLE
Cfg dataset; better assertions and comments

### PR DESCRIPTION
**Pull request by arturtoshev**
- make `dataset` a group of config parameters
- improve usability with better assertions and comments

### CODE REVIEW

File: lagrangebench/train/trainer.py [hunk 1]
```diff
@@ -253,7 +253,9 @@ def train(
         update_fn = partial(_update, loss_fn=loss_fn, opt_update=self.opt_update)
 
         # init values
-        pos_input_and_target, particle_type = next(iter(loader_train))
+        raw_batch = next(iter(loader_train))
+        raw_batch = jax.tree_map(lambda x: jnp.array(x), raw_batch)  # numpy to jax
+        pos_input_and_target, particle_type = raw_batch
```
Comment (gerkone) [inline lagrangebench/train/trainer.py:258]: did this actually have an impact?

File: README.md [hunk 2]
```diff
@@ -71,6 +71,9 @@ To run JAX on GPU, follow [Installing JAX](https://jax.readthedocs.io/en/latest/
 pip install -U "jax[cuda12]==0.4.29"
 ```
 
+> Note: as of 27.06.2024, to make our GNN models **deterministic** on GPUs, you need to set `os.environ["XLA_FLAGS"] = "--xla_gpu_deterministic_ops=true"`. However, all current models rely of `scatter_sum`, and this operation seems to be slower than running a normal for-loop in Python, when executed in deterministic mode, see [#17844](https://github.com/google/jax/issues/17844) and [#10674](https://github.com/google/jax/discussions/10674).
```
Comment (gerkone) [inline README.md:74]: good call

**Review by gerkone — APPROVED** [review]
Does not disrupt anything, so fine by me. In the end you didn't change the logic of the dataset class. Should we speity in the docstring to the dataset that download starts if the directory does not exist?
```
  dataset_path: Path to the dataset. Download will start automatically if dataset_path does not exist.
```

File: lagrangebench/train/trainer.py (hunk 1 above)
Comment (arturtoshev) [reply lagrangebench/train/trainer.py:258]: If the model is the SitL model, then yes. Without converting `particle_type` from a numpy to a jax array, SitL throws an error in the SPH solver call.

**Comment by arturtoshev** [comment]
> Does not disrupt anything, so fine by me. In the end you didn't change the logic of the dataset class. Should we speity in the docstring to the dataset that download starts if the directory does not exist?
> 
> ```
>   dataset_path: Path to the dataset. Download will start automatically if dataset_path does not exist.
> ```

there you go: [9f1f397](https://github.com/tumaer/lagrangebench/pull/31/commits/9f1f397ff141712ac407f918d90c289146f369cd)

**Review by gerkone — APPROVED** [review]

**Pull request merged**